### ISSUE TITLE
Why my code get better f1_score on older versions

### ISSUE THREAD

**Issue by wowhhh** [tensorflow/tensorflow#58431]
### Issue Type

Bug

### Source

binary

### Tensorflow Version

2.10.0 , 2.7.0, 2.4.0 

### Custom Code

No

### Python version

3.7.10

### Current Behaviour?

```shell
Hi, I adopt different tensorflow and keras versions with my code. But it return better f_1 Score on older versions. The reproduction results are as follows:
|Tensorflow|Keras|F1 Score|
|--|--|--|
|2.7.0|2.7.0|0.29846938775510207|
|2.4.0|2.4.0|0.7376928728875827|
I also test it with TensorFlow 2.10.0, and the f_1 score is 0.6031746031746031.
```

### Standalone code to reproduce the issue

```shell
I post the code on colad.[click me to reproduce](https://colab.research.google.com/drive/1BWZMI7uPmYfOgobBHRH0H2q3WXOSmdDn?usp=sharing)
```

**Comment by bhack** [comment]
Your code is not accessible.
Please can you try also with the last stable version TF 2.10?

**Comment by wowhhh** [comment]
Thanks for your quick reply! I modified the access permission. [colab code](https://colab.research.google.com/drive/1BWZMI7uPmYfOgobBHRH0H2q3WXOSmdDn?usp=sharing)

Besides, I test it with TF 2.10 and the F1 score is  0.6031746031746031.

**Comment by wowhhh** [comment]
Hi!
- [x] refactor the colab with the new imports
- [x] upload datset
[Download dataset](https://github.com/tensorflow/tensorflow/files/9929997/dataset.zip)


> Please don't import Keras directly. Import it from TF e.g.`from tensorflow.keras.models..`
> 
> Can you refactor the colab with the new imports? Also please share your input file to reproduce the results.

**Comment by bhack** [comment]
It is hard to compare results between TF version with you notebook as if you see your code gives different results with the same TF version across multiple runs.
Other then seeds you need probably to remove other random sources for a reproducible experiment. 
When you will reproduce the same results across multiple run on the same TF version we could check if there is really a numerically relevant difference.

**Comment by wowhhh** [comment]
Hello, I try to set the random seed using this approach. 
https://github.com/keras-team/keras/issues/2743#issuecomment-904169662
```python
import random
random.seed(2022) 
np.random.seed(2022)  
tf.random.set_seed(2022) 
```
I added ```shuffle = False``` in ```model.fit()```. And I executed the code five times with TF 2.10 but it gave the same result which is 0.5364372469635627. 
It' s weird why the results of your tests are inconsistent. Can you run it several times to see if there are still inconsistent results.I will also continue to test it .
Thanks!!!

**Comment by bhack** [comment]
Ok,  now I see the difference with 2.4.0.
I got very similar results also with TF 2.10 and TF Nightly (`f1_score:  0.5369807497467073`)

I think that you can reopen this ticket in the Keras repo   https://github.com/keras-team/keras.

**Comment by tilakrayal** [comment]
@wowhhh,
Thanks for opening this issue. Development of keras moved to another [repository](https://github.com/keras-team/keras/issues). 
 
 
Could you please post this issue on keras-team/keras [repo](https://github.com/keras-team/keras/issues).
To know more please refer:
https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999
Thank you!

**Comment by wowhhh** [comment]
hi, I post this issue on [keras#17217](https://github.com/keras-team/keras/issues/17217).
Thanks for your reply!

**Comment by bhack** [comment]
Thanks
